src/lib.rs

```diff
@@ -917,6 +967,37 @@ impl<T, L, R> Either<(L, T), (R, T)> {
     }
 }
 
+/// `Either<L, R>` is an iterator if both `L` and `R` are iterators.
+impl<L, R> Either<L, R>
+where
+    L: IntoIterator,
+    R: IntoIterator,
+{
+    /// Factors an `Either` of `Itorator`s to be an `Itorator` of `Either`s
+    /// ```
+    /// use either::*;
+    /// let left: Either<_, Vec<u8>> = Left(&["hello"]);
+    /// assert_eq!(left.factor_iter().next(), Some(Left(&"hello")));
+
+    /// let right: Either<&[&str], _> = Right(vec![0, 1]);
+    /// assert_eq!(right.factor_iter().collect::<Vec<_>>(), vec![Right(0), Right(1)]);
+    ///
+    /// ```
+    // TODO(MSRV): doc(alias) was stabilized in Rust 1.48
+    // #[doc(alias = "transpose")]
+    pub fn factor_iter(
```

Comment:
Why not call this `transpose`, in the fashion of [`Option::transpose`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.transpose)?

Reply:
There are multiple `factor_*` methods with this `transpose` comment already, so I don't think this `Iterator` one is more "special" to claim the name. :)

Reply:
Fair enough!